### DISCUSSION TITLE
Allow inter-broker listeners to be configured

### DISCUSSION THREAD

**Discussion by neeraj-laad** [strimzi/strimzi-kafka-operator#10080] [Ideas]
Currently listeners cannot be configured to use the ports reserved for inter-broker communication:
- Control plane listener (9090)
- Replication listener (9091)

The cluster operator creates these listeners in a specific way and there is no configurable way to influence that.  It would be useful to allow users to configure these listeners in custom resources to provide more control for some special cases.  

For example: this might also be useful to support stretch clusters where the control plane and replication need to happen across multiple Kubernetes clusters and need to be exposed outside the cluster in a specific way (Ingress/LoadBalancer etc.) or configured with custom certificates. There are possibly other scenarios where this might be useful.

Wanted to see if this has been raised/considered in the past. Also keen to see what drawbacks / issues this might introduce.

**Comment by scholzj** [comment]
There is I think an existing issue for stretch cluster. But it is a way more work than just configuring the listeners.

**Comment by neeraj-laad** [comment]
Completely agree that the stretch cluster is a much bigger piece of work. This discussion was mainly to get community view on the idea of allowing users to configure these listeners. If there are no serious objections, I would like to start a proposal specifically to allow these to be configured or overridden in specific cases.
